feat: Jira Validator Option

iammuho opened the pull request:
Issue: https://github.com/mergeability/mergeable/issues/482
Note: For this option to be working, options should be running on async. Before this PR to be accepted, please look at this PR: https://github.com/mergeability/mergeable/pull/481 (MERGED)
---

This PR adds a `Jira` option to the below validators. If anyone wants to check JIRA TICKET to be validated through JIRA APIs;

1. Enable JIRA Settings via env variables;

```
JIRA_PASSWORD //API TOKEN
JIRA_USERNAME //USERNAME
JIRA_HOST //HOST
JIRA_PROTOCOL // default 'https'
JIRA_VERSION // default '2'
JIRA_STRICT_SSL // default 'true'
```

2. Enable JIRA VALIDATION option in validators;

```
---
version: 2
mergeable:
  - when: pull_request.*, pull_request_review.*
    name: 'checkfirst'
    validate:
      - do: title
        jira:
          regex: '[A-Z][A-Z0-9]+-\d+'
          regex_flag: none
          message: 'JIRA Ticket ID is not valid'
```

You can use JIRA VALIDATION option in below validators;

- title
- headRef
- description
- label
- commit
- milestone

iammuho commented:
hey @jusx any comments for this ?

jusx commented on lib/validators/options_processor/options/jira.js:
```diff
@@ -0,0 +1,82 @@
+    let regexObj
+
+    const filter = rule.jira
+    const enabled = filter['enabled']
```
The enabled sub-option seem redundant to me. Maybe I'm missing something. Do you have a use-case where you have the `jira` option configured but you want to disable it? Wouldn't that be the same as just removing the `jira` option altogether?

jusx reviewed (commented):
Overall looks good. Left a comment about the enabled suboption.

iammuho commented:
hey @jusx , you can check again. You are right, no need for enabled.

jusx commented on lib/validators/commit.js:
```diff
@@ -17,6 +17,12 @@ class Commit extends Validator {
     ]
     this.supportedSettings = {
+      jira: {
+        enabled: 'boolean',
```
We should remove all references to `enabled` since we've agreed it's not relevant.

jusx commented on __tests__/unit/validators/options_processor/options/jira.test.js:
```diff
@@ -0,0 +1,88 @@
+  const checkTicketStatus = jest.fn().mockReturnValue(false)
+  jira.checkTicketStatus = checkTicketStatus
+
+  const rule = { jira: { enabled: true, regex: 'test' } }
```
Same as https://github.com/mergeability/mergeable/pull/483/files#r566509892 as well as in all the validators and the tests.

jusx reviewed (commented):
@iammuho Added additional comments. I would recommend including some docs either in this PR or create an issue regarding docs so that it is in the backlog.

iammuho commented:
@jusx , removed all references and sent the docs. I think in the docs we should also enable `options` menu to explain options one by one like validators and actions, because i.e JIRA option does work only self-hosted (to set environment values) For now, I've added how to add JIRA option to required validators, on another PR I'll add `options` part to the docs.
